lib/common.js

```diff
@@ -841,6 +841,73 @@ function readdir(dir, req, origOpts) {
     return (mreq);
 }
 
+function bucketList(p, req, origOpts) {
```

Comment:
The naming of this function is confusing to me. It could be the function to list things for manta buckets or it could be the function to list all the buckets for an account that uses manta buckets. Since it's the general `list` function for manta buckets buckets and objects and there is terminology overload for *bucket* with regular manta maybe it makes sense to both move and rename this. Maybe we should have a `./lib/buckets/common.js` module and put this in there and perhaps rename it to `list` or as I mentioned in a comment on the related node-boray CR we could make the decision on the type of operation here in muskie and break this down into `listObjects` and `listBuckets` functions.

Reply:
Agreed.  I'll move it into `./lib/buckets/common.js`, and then figure out how it should be dealt with wrt to having a `listObjects` and `listBuckets` function without `.search()`